osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuMovement.cs

```diff
@@ -0,0 +1,571 @@
+﻿using System;
```

Comment:
It would be nice if `ExtractMovement` could be subdivided into more functions because it spans a few hundred lines in it's current state

Reply:
It's indeed a complex function. I'm not sure if dividing it would make it more readable though.